### PR TITLE
[Upstream] - Update-istio version based on go version 1.21

### PR DESCRIPTION
**Pull request by mholder6**
**What this PR does / why we need it**:

Updated the istio libraries based on go versioning 1.21

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -74,3 +74,6 @@ docker-compose.yml
 
 # Istio installer downloaded by quick install
 istio-*
+
+#Notes on how different versions of the repo
```
Comment (spolti) [inline .gitignore]: please remove it.

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/kserve/kserve
 
-go 1.21
+go 1.21
```
Comment (spolti) [inline go.mod]: remove the space

File: pkg/apis/serving/v1beta1/openapi_generated.go [hunk 3]
```diff
@@ -1333,7 +1333,6 @@ func schema_pkg_apis_serving_v1alpha1_SupportedModelFormat(ref common.ReferenceC
 						},
 					},
 				},
-
```
Comment (spolti) [inline pkg/apis/serving/v1beta1/openapi_generated.go:1336]: remove this file from the changes.

File: pkg/apis/serving/v1beta1/openapi_generated.go [hunk 4]
```diff
@@ -1,10129 +0,0 @@
-//go:build !ignore_autogenerated
```
Comment (spolti) [inline pkg/apis/serving/v1beta1/openapi_generated.go]: Can you please remove this file from your changes?

**Review by spolti — CHANGES_REQUESTED** [review]

File: go.mod [hunk 5]
```diff
@@ -27,10 +27,10 @@ require (
 	gomodules.xyz/jsonpatch/v2 v2.4.0
 	google.golang.org/api v0.151.0
 	google.golang.org/protobuf v1.33.0
-	istio.io/api v1.19.4
+	istio.io/api v1.21.5
 	istio.io/client-go v1.19.4
-	k8s.io/api v0.28.4
-	k8s.io/apimachinery v0.28.4
+	k8s.io/api v0.29.0
+	k8s.io/apimachinery v0.29.0
```
Comment (terrytangyuan) [inline go.mod:33]: Is it intended to bump these versions as well?

Comment (mholder6) [reply go.mod:33]: I believe these automatically upgraded when running go mod tidy? After upgrading the istio libs

Comment (spolti) [reply go.mod:33]: @terrytangyuan, correct, see that we are pinning the k8s version here:
https://github.com/kserve/kserve/pull/3825/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R141

there were some incompatibilities with this new version that need closer attention.

**Review by spolti — APPROVED** [review]
/lgtm

**Comment by spolti** [comment]
@yuzisun hi, can you please take a look?

**Review by yuzisun — APPROVED** [review]
/lgtm
/approve

**Pull request merged**